### PR TITLE
add rename last file function

### PR DESCRIPTION
**Pull request by wandersoncferreira**
I came to the Issues page to request this functionality, however someone else had already done it.

You propose a `org-download-rename-last-file` and I thought it would perfectly solve my problems too.

I have implemented my own version to accomplish that and I would like to see there are anything useful to you. My Elisp is not the sharpest, but it's something.

Please, let me know if I can do anything to improve this bit of code.
Thx

### CODE REVIEW

File: org-download.el [hunk 1]
```diff
@@ -98,11 +98,17 @@ See `org-download--dir-1' for more info."
   :group 'org-download)
 (make-variable-buffer-local 'org-download-image-dir)
 
+
 (defcustom org-download-heading-lvl 0
   "Heading level to be used in `org-download--dir-2'."
   :group 'org-download)
 (make-variable-buffer-local 'org-download-heading-lvl)
 
+(defcustom org-download-path-last-file nil
```
Comment (abo-abo) [inline org-download.el]: This should be a `defvar`.

File: org-download.el [hunk 2]
```diff
@@ -330,12 +336,36 @@ It's inserted before the image link and is used to annotate it.")
                   (funcall org-download-method link))
                  (t
                   (org-download--fullname link ext)))))
+
```
Comment (abo-abo) [inline org-download.el]: Please use `M-x` `delete-trailing-whitespace`.

File: org-download.el [hunk 3]
```diff
@@ -330,12 +336,36 @@ It's inserted before the image link and is used to annotate it.")
...
         (when (eq org-download-method 'attach)
           (org-attach-attach filename nil 'none))
         (org-download-insert-link link filename)))))
 
+
+(defun org-download-rename-last-file ()
+  "Rename the last downloaded file saved in your computer."
+  (interactive)
+  (let* ((dir-path (org-download--dir))
+	 (newname (read-string "Rename last file to: "))
```
Comment (abo-abo) [inline org-download.el]: Please use `M-x` `untabify`.

File: org-download.el [hunk 4]
```diff
@@ -330,12 +336,36 @@ It's inserted before the image link and is used to annotate it.")
...
+	 (ext (file-name-extension org-download-path-last-file))
+	 (newpath (concat dir-path "/" newname "." ext)))
+    (when org-download-path-last-file
+      (rename-file org-download-path-last-file newpath 1)
+      (message "File successfully renamed...")
+      (org-download-replace-old-filepath (file-name-nondirectory org-download-path-last-file) (concat newname "." ext))
+      (setq org-download-path-last-file newpath))))
+
+(defun org-download-replace-old-filepath (oldpath newpath)
```
Comment (abo-abo) [inline org-download.el]: Could use a more generic name, like `org-download-replace-all`.

**Comment by wandersoncferreira** [comment]
Hi @abo-abo , thanks for the suggestions. I made all the changes.

**Comment by abo-abo** [comment]
Thanks.

**Pull request closed without merging**